File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let j = 1; // Initialize the count of unique elements to 1
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[j - 1]) {
            nums[j] = nums[i]; // Overwrite the next unique element
            j++;
        }
    }
    return j;
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) return 0; // Edge case: empty array
        
        int i = 0; // Pointer to track unique elements
        for (int j = 1; j < nums.size(); j++) {
            if (nums[j] != nums[i]) { // If a new unique element is found
                i++;
                nums[i] = nums[j]; // Move it to the correct position
            }
        }
        return i + 1; // Length of the modified array
    }
};
